graylog2-server/src/main/java/org/graylog2/indexer/results/FieldStatsResult.java

```diff
@@ -47,13 +48,13 @@ public FieldStatsResult(ValueCountAggregation valueCountAggregation,
         this.cardinality = cardinalityAggregation == null ? Long.MIN_VALUE : cardinalityAggregation.getCardinality();
 
         if (extendedStatsAggregation != null) {
-            sum = extendedStatsAggregation.getSum();
-            sumOfSquares = extendedStatsAggregation.getSumOfSquares();
-            mean = extendedStatsAggregation.getAvg();
-            min = extendedStatsAggregation.getMin();
-            max = extendedStatsAggregation.getMax();
-            variance = extendedStatsAggregation.getVariance();
-            stdDeviation = extendedStatsAggregation.getStdDeviation();
+            sum = Optional.ofNullable(extendedStatsAggregation.getSum()).orElse(Double.NaN);
+            sumOfSquares = Optional.ofNullable(extendedStatsAggregation.getSumOfSquares()).orElse(Double.NaN);
+            mean = Optional.ofNullable(extendedStatsAggregation.getAvg()).orElse(Double.NaN);
+            min = Optional.ofNullable(extendedStatsAggregation.getMin()).orElse(Double.NaN);
+            max = Optional.ofNullable(extendedStatsAggregation.getMax()).orElse(Double.NaN);
+            variance = Optional.ofNullable(extendedStatsAggregation.getVariance()).orElse(Double.NaN);
+            stdDeviation = Optional.ofNullable(extendedStatsAggregation.getStdDeviation()).orElse(Double.NaN);
```

Comment:
I prefer this one, because it neither requires a 3rd party library nor is it as verbose as the ternary operator in this case. What are the downsides/the "abuse" of using it?